crates/sui-framework/packages/move-stdlib/sources/macros.move

```diff
@@ -0,0 +1,71 @@
+
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This module holds shared implementation of macros used in `std`
+module std::macros {
+    public(package) macro fun num_max($x: _, $y: _): _ {
+        let x = $x;
+        let y = $y;
+        if (x > y) x
+        else y
+    }
+
+    public(package) macro fun num_min($x: _, $y: _): _ {
+        let x = $x;
+        let y = $y;
+        if (x < y) x
+        else y
+    }
+
+    public(package) macro fun num_diff($x: _, $y: _): _ {
+        let x = $x;
+        let y = $y;
+        if (x > y) x - y
+        else y - x
+    }
+
+    public(package) macro fun num_divide_and_round_up($x: _, $y: _): _ {
+        let x = $x;
+        let y = $y;
+        if (x % y == 0) x / y
+        else x / y + 1
```

Comment:
Would this be more efficient? Whatever that means for us:

```suggestion
        if (x == 0) x
        else (x - 1) / y + 1
```

Reply:
(I see now that this is a copy of what we had before -- so fine to disregard as a comment for this PR and tackle it separately).